code_generator/GdkCodeGenerator/Templates/UnityComponentDataGenerator.tt

```diff
@@ -14,6 +14,8 @@ namespace <#= qualifiedNamespace #>
 { 
     public struct <#= componentDetails.TypeName #> : IComponentData, ISpatialComponentData
     {
+        public uint ComponentId => <#= componentDetails.ComponentId #>;
```

Comment:
can presumably make this const

Reply:
Ah just seen this is a property.

Reply:
Alternatively we can use something similar to dynamic. Would rather not make things look like fields by default when we don't have to.

Reply:
We could use something like dynamic for sure - that could be a later refactor I think though. Will need to resolve this with the approach monobehaviours took too